bot/cogs/bot.py

```diff
@@ -75,40 +75,54 @@ def __init__(self, bot: AutoShardedBot):
+                if content.lower().startswith("python"):
+                    content = content[6:]
+                elif content.lower().startswith("py"):
+                    content = content[2:]
+                # Strip again to remove the whitespace(s) left before the code
+                # If the msg looked like "Python <code>" before removing Python
+                content = content.strip()
+                return content
+
     async def on_message(self, msg: Message):
         if msg.channel.id in self.previous_format_times:
             if time.time()-self.previous_format_times[msg.channel.id] > 300 or msg.channel.id == DEVTEST_CHANNEL:
-                if msg.content.count("\n") >= 3:
                     try:
-                        # Filtering valid Python codeblocks and exiting if a valid Python codeblock is found
-                        if re.match(msg.content, "```(python|py)\n((?:.*\n*)+)```", re.IGNORECASE):
-                            return
-                        else:
-                            # Format the block into python source
-                            content = ""
-                            for line in msg.content.split("\n"):
-                                content += line.strip("`") + "\n"
-                            # Remove "Python" or "Py" from top of the message if exists
-                            if content.startswith("python") or content.startswith("Python"):
-                                content = content[6:]
-                            elif content.startswith("py") or content.startswith("Py"):
-                                content = content[2:]
-                            # Strip again to remove the whitespace(s) left if the msg looks like
-                            # " Python"
-                            content = content.strip()
-
                         # Attempts to parse the message into an AST node.
                         # Invalid Python code will raise a SyntaxError.
+                        content = self.codeblock_stripping(msg.content)
+                        if not content:
+                            return
+
                         tree = ast.parse(content)
 
                         # Multiple lines of single words could be interpreted as expressions.
                         # This check is to avoid all nodes being parsed as expressions.
                         # (e.g. words over multiple lines)
                         if not all(isinstance(node, ast.Expr) for node in tree.body):
-                            header = (f"Hey {msg.author.mention} I noticed you tried to paste code. ",
-                                      "Here's how you should have done it:")
-                            information = Embed(title="Codeblocks", description=get_tag_data("codeblock"))
-                            await msg.channel.send(header, embed=information)
+                            howto = (f"Hey {msg.author.mention} I noticed you tried to paste code. ",
```

Comment:
You didn't change this message yet, and it's still neither grammatically correct nor expressed in the right tone. The bot should have a whimsical yet polite tone. This is missing a period, as well.  Please see my earlier comment that was dismissed by gdude even though it wasn't addressed :stuck_out_tongue: